### ISSUE TITLE
Build failed with latest indexmap (2.0.0)

### ISSUE THREAD

**Issue by jcamiel** [eminence/xmltree-rs#39]
Hello, 

First of all, thank you for you crate.
When using the latest crate of indexmap (2.0.0) and xmltree 0.10.3, I've a build failure:

Cargo.toml:

```toml
[package]
name = "xmltreetest"
version = "0.1.0"
edition = "2021"

[dependencies]
xmltree = { version = "0.10.3",  features = ["attribute-order"] }
indexmap = "2.0.0"
```

main.rs:

```rust
use indexmap::IndexMap;
use xmltree::Element;

fn main() {
    let element = Element {
        prefix: None,
        namespace: None,
        namespaces: None,
        name: "foo".to_string(),
        attributes: IndexMap::new(),
        children: vec![],
    };
    println!("{element:?}");
}
```

Build:
```shell
$ cargo build
$    Compiling xmltreetest v0.1.0 (/Users/jc/Documents/Dev/xmltreetest)
error[E0308]: mismatched types
  --> src/main.rs:10:21
   |
10 |         attributes: IndexMap::new(),
   |                     ^^^^^^^^^^^^^^^ expected `IndexMap<String, String>`, found `IndexMap<_, _>`
   |
   = note: `IndexMap<_, _>` and `indexmap::map::IndexMap<String, String>` have similar names, but are actually distinct types
note: `IndexMap<_, _>` is defined in crate `indexmap`
  --> /Users/jc/.cargo/registry/src/index.crates.io-6f17d22bba15001f/indexmap-2.0.0/src/map.rs:81:1
   |
81 | pub struct IndexMap<K, V, S = RandomState> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
note: `indexmap::map::IndexMap<String, String>` is defined in crate `indexmap`
  --> /Users/jc/.cargo/registry/src/index.crates.io-6f17d22bba15001f/indexmap-1.9.3/src/map.rs:71:1
   |
71 | pub struct IndexMap<K, V, S = RandomState> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   = note: perhaps two different versions of crate `indexmap` are being used?

For more information about this error, try `rustc --explain E0308`.
error: could not compile `xmltreetest` (bin "xmltreetest") due to previous error
```

If I switch to indexmap 1.9.3:

```shell
$ cg build
    Finished dev [unoptimized + debuginfo] target(s) in 0.01s
```

**Comment by waywardmonkeys** [comment]
You need to use the same version as `xmltree` (currently 1.x). I've submitted a PR to update to v2: #45 

Once a new release is out, you can also use `AttributeMap` rather than having to know the type yourself as that has been made into a public type alias since the last release.

**Comment by eminence** [comment]
This issue has been fixed in the newly released v0.11.0 version.  Thanks to @waywardmonkeys for the fix, and to everyone for your patience
